src/test/llmnode_test.cpp

```diff
@@ -0,0 +1,290 @@
+#include <thread>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+#include <openvino/openvino.hpp>
+#include <pybind11/embed.h>
+
+#include "../config.hpp"
+#include "../dags/pipelinedefinition.hpp"
+#include "../grpcservermodule.hpp"
+#include "../http_rest_api_handler.hpp"
+#include "../kfs_frontend/kfs_graph_executor_impl.hpp"
+#include "../kfs_frontend/kfs_grpc_inference_service.hpp"
+#include "../mediapipe_internal/mediapipefactory.hpp"
+#include "../mediapipe_internal/mediapipegraphdefinition.hpp"
+#include "../mediapipe_internal/mediapipegraphexecutor.hpp"
+#include "../metric_config.hpp"
+#include "../metric_module.hpp"
+#include "../model_service.hpp"
+#include "../precision.hpp"
+#include "../python/pythoninterpretermodule.hpp"
+#include "../python/pythonnoderesources.hpp"
+#include "../servablemanagermodule.hpp"
+#include "../server.hpp"
+#include "../shape.hpp"
+#include "../stringutils.hpp"
+#include "../tfs_frontend/tfs_utils.hpp"
+#include "c_api_test_utils.hpp"
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#include "mediapipe/framework/calculator_graph.h"
+#include "mediapipe/framework/calculator_runner.h"
+#pragma GCC diagnostic pop
+
+#include "opencv2/opencv.hpp"
+#include "python/python_backend.hpp"
+#include "test_utils.hpp"
+
+using namespace ovms;
+
+class LLMFlowTest : public ::testing::TestWithParam<std::string> {
+protected:
+    ovms::Server& server = ovms::Server::instance();
+
+    const Precision precision = Precision::STRING;
+    std::unique_ptr<std::thread> t;
+    std::string port = "9178";
+
+    void SetUpServer(const char* configPath) {
+        ::SetUpServer(this->t, this->server, this->port, configPath);
+    }
+
+    void SetUp() override {
+    }
+    void TearDown() {
+        server.setShutdownRequest(1);
+        t->join();
+        server.setShutdownRequest(0);
+    }
+};
+
+class LLMFlowKfsTest : public LLMFlowTest {
+public:
+    void SetUp() {
+        SetUpServer("/ovms/src/test/llm/config_llm_dummy_kfs.json");
+    }
+};
+
+// --------------------------------------- OVMS LLM nodes tests
+// Test disabled by default - needs LLM models to work in /workspace directory:
+// openvino_detokenizer.bin  openvino_detokenizer.xml  openvino_model.bin  openvino_model.xml  openvino_tokenizer.bin  openvino_tokenizer.xml
+TEST_F(LLMFlowKfsTest, DISABLED_Infer) {
+    const ovms::Module* grpcModule = server.getModule(ovms::GRPC_SERVER_MODULE_NAME);
+    KFSInferenceServiceImpl& impl = dynamic_cast<const ovms::GRPCServerModule*>(grpcModule)->getKFSGrpcImpl();
+    ::KFSRequest request;
+    ::KFSResponse response;
+    const std::string modelName = "llmDummyKFS";
+    request.Clear();
+    response.Clear();
+    std::vector<std::string> requestData1{"What is OpenVINO?"};
+    std::string expectedResponse = "\n\nOpenVINO is an open-source software library for deep learning inference that is designed to optimize and run deep learning models on a variety";
```

Comment:
Please add a comment for which model exactly this is the expected output

Reply:
Model that was used: 
https://huggingface.co/meta-llama/Llama-2-7b-chat-hf
converted with optimum cli